Sources/SwiftInspectorVisitors/TypeDescription.swift

```diff
@@ -366,12 +367,19 @@ extension TypeSyntax {
         isType: typeIdentifier.typeOrProtocol.text == "Type")
 
     } else if let typeIdentifier = self.as(AttributedTypeSyntax.self) {
-      return .attributed(
-        typeIdentifier.baseType.typeDescription,
-        specifier: typeIdentifier.specifier?.text,
-        attributes: typeIdentifier.attributes?.compactMap {
-          $0.as(AttributeSyntax.self)?.attributeName.text
-        })
+      let specifier = typeIdentifier.specifier?.text
+      let attributes = typeIdentifier.attributes?.compactMap {
+        $0.as(AttributeSyntax.self)?.attributeName.text
+      }
+      if specifier == nil && attributes == nil {
+        assertionFailureOrPostNotification("Encountered an attributed type with no attributes or specifiers")
+        return typeIdentifier.baseType.typeDescription
```

Comment:
Hmm if we actually hit this code path it seems to indicate that we've found potentially valid syntax we didn't expect, right? I wonder if it makes sense to keep propagating `.attributed(...)` to ensure that we persist the data even if we may not correctly generate source from it today? Curious to learn more about this change.

Reply:
I think this would only happen if properties beyond `specifier` and `attributes` were added to this type.

We can keep propagating `.attributed` here. I changed to propagating the `baseType` because I figured our `asSource` would produce the same result and we should consolidate.

I'll make this change to keep propagating `.attributed` here, and in that commit I think I'll also remove the assert in `asSource` since we'll have asserted here when we hit the error case (and since the `asSource` code will now be expected if we've hit this assert). It feels better to assert here because we know we'll hit this code if we parse it, and who knows if someone will call `asSource` on a type.

Reply:
3a6712a

Reply:
> I think this would only happen if properties beyond `specifier` and `attributes` were added to this type.

I hadn't considered this possibility.

Change LGTM.